src/core/basicfb.cpp

```diff
@@ -195,12 +204,16 @@ void CBasicFB::traceInstanceData() {
     internals_c_str[i] = valueString.c_str();
   }
 
-  for(TPortId i = 0; i < internalFbs.size(); ++i) {
-    CFunctionBlock *value = getInternalFB(i);
-    std::string &valueString = internalFbs[i];
-    valueString.reserve(value->getToStringBufferSize());
-    value->toString(valueString.data(), valueString.capacity());
-    internalFbs_c_str[i] = valueString.c_str();
+  TPortId i = 0;
+  for(TFBContainerList::iterator it(getChildren().begin()); it != getChildren().end(); ++it){
+    if((*it)->isFB()) {
```

Comment:
A basic FB can only contain internal FBs (or adapters), both of which happen to return `true` for `isFB()`, so I would suggest to simply use `getChildren()` as-is.